src/main/java/com/flightstats/hub/metrics/JVMMetricsService.java

```diff
@@ -0,0 +1,118 @@
+package com.flightstats.hub.metrics;
+
+import com.flightstats.hub.app.HubProperties;
+import com.flightstats.hub.app.HubVersion;
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.MetricFilter;
+import com.codahale.metrics.jvm.GarbageCollectorMetricSet;
+import com.codahale.metrics.jvm.CachedThreadStatesGaugeSet;
+import com.codahale.metrics.jvm.MemoryUsageGaugeSet;
+import com.codahale.metrics.ScheduledReporter;
+import metrics_influxdb.InfluxdbProtocol;
+import metrics_influxdb.InfluxdbReporter;
+import metrics_influxdb.HttpInfluxdbProtocol;
+import metrics_influxdb.UdpInfluxdbProtocol;
+import com.google.inject.Singleton;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.inject.Inject;
+import org.slf4j.LoggerFactory;
+import org.slf4j.Logger;
+import java.util.SortedSet;
+import java.util.concurrent.TimeUnit;
+import static com.flightstats.hub.app.HubHost.getLocalName;
+
+@Singleton
+public class JVMMetricsService {
+    private final Logger logger = LoggerFactory.getLogger(JVMMetricsService.class);
+
+    @Inject
+    private MetricRegistry registry;
+
+    @Inject
+    private HubVersion hubVersion;
+
+    private ScheduledReporter influxReporter;
+    private final String protocolScheme = HubProperties.getProperty("influx.protocolScheme", "http");
```

Comment:
I'd probably need to fiddle with it a little bit, but my gut instinct is that it might make sense to pull all the configuration for this out into another class and (again) inject it. That'd clear up responsibilities a little bit. Any time I use `@VisibleForTesting` I tend to wonder if my class is doing too much.

Reply:
sounds good, I'll give that a shot

Reply:
Agreed. One of the paradigms I was trying to move towards was having a POJO that represented configuration and then inject it to those that needed it.

We should probably mark the `HubProperties` class as deprecated as a reminder that its not going to live through the future.

Reply:
I almost went on a tangent about how I wish we'd get rid of `HubProperties` and thought better of it. 😃